src/db-up/azext_db_up/_validators.py

```diff
@@ -30,43 +30,59 @@ def db_down_namespace_processor(db_type):
     return lambda cmd, namespace: _process_db_down_namespace(namespace, db_type=db_type)
 
 
+# pylint: disable=bare-except
 def _process_db_up_namespace(cmd, namespace, db_type=None):
-    # populate from cache if existing
-    _set_value(db_type, namespace, 'location', 'location', cache=False)
-    _set_value(db_type, namespace, 'resource_group_name', 'group', cache=False)
+    # populate from cache if existing when no resource group name provided
+    resource_client = resource_client_factory(cmd.cli_ctx)
+    if namespace.resource_group_name is None:
+        _set_value(db_type, namespace, 'resource_group_name', 'group', cache=False)
+        try:
+            resource_client.resource_groups.get(namespace.resource_group_name)
+        except:  # Clear resource group name information when it is invalid
+            namespace.resource_group_name = None
 
-    # generate smart defaults
+    # populate from cache if existing when no location provided
+    if namespace.location is None:
+        _set_value(db_type, namespace, 'location', 'location', cache=False)
+    # generate smart defaults when namespace.location is None
     if _get_value(db_type, namespace, 'location', 'location') is None:
         try:
             get_default_location_from_resource_group(cmd, namespace)
         except (CLIError, ValidationError):
-            namespace.location = 'westus2'
-    _set_value(db_type, namespace, 'location', 'location', namespace.location)
+            namespace.location = 'eastus'
+    _set_value(db_type, namespace, 'location', 'location', default=namespace.location)
+
+    # When resource group name in namespace is different from what in cache, reset it and create new server name
+    if namespace.resource_group_name != get_config_value(db_type, 'group', None):
+        set_config_value(db_type, 'group', namespace.resource_group_name)
+        if namespace.server_name is None:
+            namespace.server_name = create_random_resource_name('server')
+            set_config_value(db_type, 'server', namespace.server_name)
 
+    # When no resource group name in namespace and cache, create new resource group with random name
     create_resource_group = True
-    resource_client = resource_client_factory(cmd.cli_ctx)
-    if namespace.resource_group_name is None:
+    if namespace.resource_group_name is None:  # No resource group provided and in cache
         namespace.resource_group_name = create_random_resource_name('group')
     else:
         try:
             resource_client.resource_groups.get(namespace.resource_group_name)
             create_resource_group = False
-        except CloudError:
-            pass
+        except CloudError as ex:  # throw exception when resource group name is invalid
+            raise ex
```

Comment:
Below is the logic of create resource group. The original logic was that if the resource group was not found in the Service, the resource group was created on the Service and the rest of the process was followed.
Is it in line with the expectation to directly exit the following process?